commands/dns.go

```diff
@@ -41,20 +40,56 @@ func (cmd *DNS) Commands() []cli.Command {
 func (cmd *DNS) Run(c *cli.Context) error {
 	cmd.out.Info.Println("Configuring DNS")
 
-	if util.SupportsNativeDocker() {
-		cmd.ConfigureDNS(cmd.machine, c.String("nameservers"))
-	} else if cmd.machine.IsRunning() {
-		cmd.ConfigureDNS(cmd.machine, c.String("nameservers"))
-		cmd.ConfigureRoutes(cmd.machine)
-	} else {
+	if !util.SupportsNativeDocker() && !cmd.machine.IsRunning() {
 		return cmd.Error(fmt.Sprintf("Machine '%s' is not running.", cmd.machine.Name), "MACHINE-STOPPED", 12)
 	}
 
+	if err := cmd.StartDNS(cmd.machine, c.String("nameservers")); err != nil {
+		return cmd.Error(err.Error(), "DNS-SETUP-FAILED", 13)
+	}
+
+	if !util.SupportsNativeDocker() {
+		cmd.ConfigureRoutes(cmd.machine)
+	}
 	return cmd.Success("DNS Services have been started")
 }
 
-// RemoveHostFilter removs the host filter from the xhyve bridge interface
-func (cmd *DNS) RemoveHostFilter(ipAddr string) {
+// ConfigureRoutes will configure routing to allow access to containers on IP addresses
+// within the Docker Machine bridge network
+func (cmd *DNS) ConfigureRoutes(machine Machine) {
+	cmd.out.Info.Println("Setting up local networking (may require your admin password)")
+
+	if util.IsMac() {
+		cmd.configureMacRoutes(machine)
+	} else if util.IsWindows() {
+		cmd.configureWindowsRoutes(machine)
+	}
+}
+
+// ConfigureMac configures DNS resolution and network routing
+func (cmd *DNS) configureMacRoutes(machine Machine) {
+	cmd.out.Verbose.Print("Configuring network routing for macOS")
```

Comment:
Is this log valuable, as it directly follows this info log
> Setting up local networking (may require your admin password)

Reply:
Maybe not I can remove those.